test/net_http_test.rb

```diff
@@ -511,42 +492,15 @@ def close_circuit!(hostname: SemianConfig['toxiproxy_upstream_host'], toxic_port
 
   def with_server(ports: [SemianConfig['http_port']], reset_semian_state: true)
     ports.each do |port|
-      hostname = '0.0.0.0'
-      begin
-        server = nil
-        server_threw_error = false
-        server_thread = Thread.new do
-          Thin::Logging.silent = true
-          server = Thin::Server.new(hostname, port, RackServer)
-          begin
-            server.start
-          rescue StandardError
-            server_threw_error = true
-            raise
-          end
-        end
-
-        begin
-          poll_until_ready(hostname: hostname, port: port)
-        rescue RuntimeError
-          server_thread.kill
-          server_thread.join if server_threw_error
-          raise
-        end
-
-        assert(server.running?)
-        reset_semian_resource(hostname: SemianConfig['toxiproxy_upstream_host'], port: port) if reset_semian_state
+      if port == SemianConfig['http_port'] || SemianConfig['http_port_2'] # rubocop:disable all
```

Comment:
Previously, we'd reset regardless of port. Now we only reset if it's one of these two ports. Was this necessary to get things working?

My suspicion is that this is due to the fact that we only know, with certainty, that these are the two ports started. If that's the case, again we're implicitly coupled to other code (specifically, the code that starts the servers in `test/test_helper.rb`). Is there any way we can prevent this coupling?

Reply:
Previously we would start and shutdown a server on every call to `with_server`. 
And also reset the semians on them. 

These were the only two ports we'd start a server on, so now instead of spinning up a new server I just spin them beforehand and only reset the semians in `with_server`.

Reply:
I could make the `http_port` in the yaml an array, spin up server for each in `test/test_helper.rb` and do check for the ports here. 

Arguably we can get rid of this if clause and reset semian resources if we make  `http_port` in the yaml and array

Reply:
It does just work if I take out the `if` clause.